galley/pkg/fs/fssource_test.go

```diff
@@ -326,16 +327,17 @@ func TestFsSource_DeletePartResorceInFile(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Unexpected error: %v", err)
 	}
-
-	log := logChannelOutput(ch, 4)
+	//In some OS, it will trigger file deletion event
+	donec := make(chan bool)
 	expected := "[Event](Deleted: [VKey](type.googleapis.com/istio.policy.v1beta1.Rule:some.mixer.rule @v0))"
-
-	if log != expected {
-		t.Fatalf("Event mismatch:\nActual:\n%s\nExpected:\n%s\n", log, expected)
+	go checkEventOccurs(expected, ch, donec)
+	select {
+	case <-time.After(time.Second):
```

Comment:
second might be too short for CI environments to cause flakes. Can you increase the timeout here (i.e. 5 seconds)

Reply:
Alternatively, you could use a [fake watcher](https://github.com/istio/istio/issues/7877#issuecomment-425594503) and provide strong timing guarantees around event handling.

Reply:
Done